macosx/Controller.mm

```diff
@@ -341,18 +341,28 @@ + (void)initialize
     //cover our asses
     if ([NSUserDefaults.standardUserDefaults boolForKey:@"WarningLegal"])
     {
-        NSAlert* alert = [[NSAlert alloc] init];
-        [alert addButtonWithTitle:NSLocalizedString(@"I Accept", "Legal alert -> button")];
-        [alert addButtonWithTitle:NSLocalizedString(@"Quit", "Legal alert -> button")];
-        alert.messageText = NSLocalizedString(@"Welcome to Transmission", "Legal alert -> title");
-        alert.informativeText = NSLocalizedString(
-            @"Transmission is a file-sharing program."
-             " When you run a torrent, its data will be made available to others by means of upload."
-             " You and you alone are fully responsible for exercising proper judgement and abiding by your local laws.",
-            "Legal alert -> message");
-        alert.alertStyle = NSAlertStyleInformational;
-
-        if ([alert runModal] == NSAlertSecondButtonReturn)
+        auto bundleURL = (__bridge CFURLRef)[NSBundle.mainBundle bundleURL];
+        auto iconURL = (__bridge CFURLRef)[NSBundle.mainBundle URLForImageResource:@"AppIcon"];
+        auto alertMessage = CFSTR(
```

Comment:
I should leave the proper review to people who know macOS' API better than me, but just at first glance it looks like this discards the l10n. Is this change made because l10n is handled by the `bundleURL` param to `CFUserNotificationDisplayAlert()`?

Reply:
Exactly, it handles l10n. By using the bundle URL, the correct language is automatically selected. I tested it and it works as expected.